ttnn/cpp/ttnn/operations/eltwise/binary/binary_l1_interface.hpp

```diff
@@ -3,40 +3,45 @@
 #include <cstdint>
 #include <iostream>
 #include <optional>
-#include <vector>
 #include <tuple>
+#include <vector>
+#include "impl/buffers/buffer.hpp"
+#include "ttnn/tensor/types.hpp"
 
 // forward declarations
-namespace tt
-{
-namespace tt_metal
-{
+namespace tt {
+namespace tt_metal {
 enum class DataType;
 enum class Layout;
 struct MemoryConfig;
 struct ShardSpec;
-} // namespace tt_metal
-} // namespace tt
+}  // namespace tt_metal
+}  // namespace tt
 
 namespace ttnn {
-    namespace types {
-        struct Shape;
-    }
-} // namespace ttnn
-
-using EltwiseOpParams = std::tuple<ttnn::types::Shape, tt::tt_metal::DataType, tt::tt_metal::Layout, tt::tt_metal::MemoryConfig>;
+namespace types {
+struct Shape;
+}
+}  // namespace ttnn
 
+using EltwiseOpParams =
+    std::tuple<ttnn::types::Shape, tt::tt_metal::DataType, tt::tt_metal::Layout, tt::tt_metal::MemoryConfig>;
 
 // these should go to a common tensor_l1_interface header or something
 uint32_t calculate_circular_buffer_l1_allocation_size_per_core(
     const ttnn::types::Shape& shape,
     const tt::tt_metal::DataType data_type,
     const tt::tt_metal::Layout& layout,
-    const tt::tt_metal::MemoryConfig& memory_config);
-
-inline uint32_t calculate_circular_buffer_l1_allocation_size_per_core(EltwiseOpParams input)
-{
-    return calculate_circular_buffer_l1_allocation_size_per_core(std::get<ttnn::types::Shape>(input), std::get<tt::tt_metal::DataType>(input), std::get<tt::tt_metal::Layout>(input), std::get<tt::tt_metal::MemoryConfig>(input));
+    const tt::tt_metal::MemoryConfig& memory_config,
+    const uint32_t max_block_size);
+
+inline uint32_t calculate_circular_buffer_l1_allocation_size_per_core(EltwiseOpParams input, uint32_t max_block_size) {
+    return calculate_circular_buffer_l1_allocation_size_per_core(
+        std::get<ttnn::types::Shape>(input),
+        std::get<tt::tt_metal::DataType>(input),
+        std::get<tt::tt_metal::Layout>(input),
+        std::get<tt::tt_metal::MemoryConfig>(input),
+        max_block_size);
```

Comment:
would it be feasible to calculate max_block_size using provided MemoryConfig within function and remove it from the argument? if yes, that would be my guidance in order to keep interface simpler.